### ISSUE TITLE
[Feature]Display used weight vs total weight in UI / Fix Shops to follow esx weight

### ISSUE THREAD

**Issue by jigglyblunt** [DiscworldZA/gta-resources#214]
**Is your feature request related to a problem? Please describe.**
When using F2 it would be nice to see weight/weight total. When we use the shops you can drag more then your weight allows into your inventory.

**Describe the solution you'd like**
A weight/total weight in UI, :) and fix shops to support player weight.

**Describe alternatives you've considered**
In the disc-inventoryhud\server\inventory, I'm not 100% sure but if we took this code,

**Here:**

if data.destinationTier.name == 'player' then
                    data.originItem.block = true
                    local destinationPlayer = ESX.GetPlayerFromIdentifier(data.destinationOwner)
                    TriggerEvent('disc-inventoryhud:notifyImpendingAddition', data.originItem, data.originItem.qty, destinationPlayer.source)
                    destinationPlayer.addInventoryItem(data.originItem.id, data.originItem.qty)
                end

**Change to:**

if data.destinationTier.name == 'player' then
                    data.originItem.block = true
                    local destinationPlayer = ESX.GetPlayerFromIdentifier(data.destinationOwner)
                    TriggerEvent('disc-inventoryhud:notifyImpendingAddition', data.originItem, data.originItem.qty, destinationPlayer.source)
                   if destinationPlayer.canCarryItem(data.originItem.id, data.originItem.qty) then
                  destinationPlayer.addInventoryItem(data.originItem.id, data.originItem.qty)
                end
               end
                

**Additional context**
I hope this makes sense, i may try and change the coding and test it but not sure as im still learning. I appreciate you looking into this and thanks for taking the time to consider this request.

**Comment by rex2630** [comment]
Yeah, that should work, but sice ESX reads all of it from user_inventory table u will have lot problems with that

**Comment by jigglyblunt** [comment]
True, but wouldnt putting this check with disc inventory prevent you from dragging if over if ESX is detecting your player already has met the set weight? Disc Inventory right now has no way of knowing if a player is at max weight.
Appreciate you response,
-JB

**Comment by rex2630** [comment]
We will surely figure the solution. We need to add own weight system or make it support ESX weight, but that will need ESX edits from user.

**Comment by jigglyblunt** [comment]
I know you guys will figure something out, this mod is well done and you guys have the skills. Thanks for working on it and getting back to me.
-JB

**Comment by DiscworldZA** [comment]
Ready esx weight is certainly possible. Hooking into esx weight is something I considered doing instead of making my own

**Comment by jigglyblunt** [comment]
I think the esx weight would be good, but your boss any support with it is amazing, appreciate your efforts the UI is great.
-JB

**Comment by custardyfingers** [comment]
Has the weight esx weight system been implemented yet? Would be a huge plus.
